ingest/src/main/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClient.java

```diff
@@ -112,15 +160,50 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
         streamSourceInfo.validate();
         ingestionProperties.validate();
 
-        if (streamSourceInfo.isLeaveOpen()) {
-            throw new UnsupportedOperationException("LeaveOpen can't be true in ManagedStreamingIngestClient");
+        UUID sourceId = streamSourceInfo.getSourceId();
+        if (sourceId == null) {
+            sourceId = UUID.randomUUID();
+        }
+
+        byte[] streamingBytes;
+
+        try {
+            streamingBytes = IOUtils.readNBytes(streamSourceInfo.getStream(), MAX_STREAMING_SIZE_BYTES + 1);
+        } catch (IOException e) {
+            throw new IngestionClientException("Failed to read from stream.", e);
         }
-        streamSourceInfo.setLeaveOpen(true);
 
+        ByteArrayInputStream byteArrayStream = new ByteArrayInputStream(streamingBytes);
```

Comment:
I don't see how closing is being managed, but if not fully managed, we probably want this to be within a `try-with-resources` block.

Reply:
ByteArrayInputStream's close is a no-op:
https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayInputStream.html

Maybe I'll add a comment?

Reply:
Ah, thanks for explaining. Yeah, I think a comment would help make it clear.